File: ByLanguage/android/Cookbook/ListActivity/app/src/main/java/com/tq/listactivity/MainActivity.java
package com.tq.listactivity;

import android.app.ListActivity;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class MainActivity extends ListActivity {

        private static final String[] Choices = new String[]{
                "Open Website Example",
                "Open Contacts",
                "Open Phone Dialer Example",
                "Search Google Example",
                "Start Voice Command"
        };

        private final String searchTerms = "superman";

        protected void onCreate(Bundle savedInstanceState) {
                super.onCreate(savedInstanceState);

                setListAdapter(new ArrayAdapter<String>(this,
                                                        android.R.layout.simple_list_item_1,
                                                        Choices));
                getListView().setChoiceMode(ListView.CHOICE_MODE_SINGLE);
                getListView().setTextFilterEnabled(true);
                getListView().setOnItemClickListener(new AdapterView.OnItemClickListener() {
                                public void onItemClick(AdapterView<?> arg0, View view, int position, long x) {
                                        
                                }
                        });
        }

}

File: alarm/app/src/main/java/com/tq/alarm/MainActivity.kt
package com.tq.alarm

import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.viewpager.widget.ViewPager
import androidx.appcompat.app.AppCompatActivity
import com.tq.alarm.adapters.TabPagerAdapter

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tabPagerAdapter = TabPagerAdapter(this, supportFragmentManager)
        val viewPager: ViewPager = findViewById(R.id.view_pager)
        viewPager.adapter = tabPagerAdapter
        val tabs: TabLayout = findViewById(R.id.tabs)
        tabs.setupWithViewPager(viewPager)
    }
}